### PR TITLE
feat(SCT-918): timeline improvements

### PR DESCRIPTION
**Pull request by jhackett1**
small improvements to the timeline:
- show a title in all cases, rather than defaulting to "view"
- for **OLDER CASE NOTES ONLY**, show the title or description, if it's available (doing this for newer submissions needs api changes)
- for **OLDER CASE NOTES ONLY**, search case note content as well as name and author
- handle the datetime format returned by google docs
- show a more prominent banner if a person is restricted

Screenshot 2021-07-22 at 11 42 57

Screenshot 2021-07-22 at 13 05 30

### CODE REVIEW

File: components/Cases/CaseLink.tsx [hunk 1]
```diff
@@ -21,10 +21,10 @@ const getWarningNoteDetailsPageName = (form_name: string) => {
   }
 };
 
-const getLink = (
+export const getLink = (
```
Comment (JohnFarrellDev) [inline components/Cases/CaseLink.tsx:24]: this function should be unit tested

File: components/NewPersonView/Event.tsx [hunk 2]
```diff
@@ -24,7 +26,9 @@ const Event = ({ event }: Props): React.ReactElement => {
     String(event?.dateOfEvent || event?.caseFormTimestamp)
   );
 
-  const displayName = event?.formName || event?.caseFormData?.form_name_overall;
+  const displaySnippet = event?.caseFormData?.case_note_title
+    ? `${event?.caseFormData?.case_note_title}: ${event?.caseFormData?.case_note_description}`
+    : false;
```
Comment (JohnFarrellDev) [inline components/NewPersonView/Event.tsx:31]: Is this tested?

File: components/NewPersonView/Event.tsx [hunk 3]
```diff
@@ -33,21 +37,20 @@ const Event = ({ event }: Props): React.ReactElement => {
...
       <h3 className="lbh-heading-h3">
-        <CaseLink
-          formName={event.formName}
-          externalUrl={event.caseFormUrl}
-          caseFormData={event.caseFormData}
-          recordId={event.recordId}
-          personId={event.personId}
-        >
-          {displayName}
-        </CaseLink>
```
Comment (JohnFarrellDev) [inline components/NewPersonView/Event.tsx:44]: Do we even need the CaseLink component now, it only exists in the <CaseTable> component which doesn't seem to be used anywhere?

File: components/NewPersonView/Layout.spec.tsx [hunk 4]
```diff
@@ -115,4 +119,20 @@ describe('Layout', () => {
     );
     expect(screen.queryByText('Timeline')).toBe(null);
   });
+
+  it('tells unauthorised users when a resident is restricted', () => {
```
Comment (JohnFarrellDev) [inline components/NewPersonView/Layout.spec.tsx:123]: Same test but for when canManageCases(user, person) returns true.

File: utils/date.ts [hunk 5]
```diff
@@ -73,6 +73,9 @@ export const normaliseDateToISO = (str: string): string => {
     return `${year}-${month}-${days}T${hours.padStart(2, '0')}:${minutes}${
       seconds ? `:${seconds}` : ''
     }`;
+  } else if (/[0-9]{1,2}\/[0-9]{1,2}\/[0-9]{4}/.test(str)) {
```
Comment (JohnFarrellDev) [inline utils/date.ts:76]: Probably want a test for this too

**Review by JohnFarrellDev — APPROVED** [review]
I reckon the code is fine but would like to see some more tests (who knows maybe writing the tests will expose some minor oversights/bugs too)

File: components/NewPersonView/Event.tsx (hunk 3 above)
Comment (jhackett1) [reply components/NewPersonView/Event.tsx:44]: it's still used in a few places - i am keen to deprecate it

**Pull request merged**